src/utils/utils_unix.go

```diff
@@ -60,11 +60,16 @@ func getSockaddrByName(name string) syscall.Sockaddr {
 }
 
 func BindToInterface(name string) func(network, address string, conn syscall.RawConn) error {
+	sockAddr := getSockaddrByName(name)
```

Comment:
ah, right, although this has to be done in the returned function to work in case network changes (i.e. when you're on wifi)

I'll address myself

Reply:
properly fixed in https://github.com/Arriven/db1000n/commit/9924a9d84168c5b3d112c3c3796b5c5ad6ad3dac